### ISSUE TITLE
Feature Request: add avx512skx-i32x4 / avx512skx-i64x4 targets

### ISSUE THREAD

**Issue by mindbeast** [ispc/ispc#2171]
ISPC does not currently offer a target for applications that wish to take advantage of all the 32 vector registers / mask registers introduced with avx512, use double precision arithmetic (or other 64 bit types), and want to use 256 bit vector width (because of the [well-known frequency penalty](https://travisdowns.github.io/blog/2020/08/19/icl-avx512-freq.html) when using full 512 bit vectors w/heavy simd operations).

A avx512skx-i32x4 / avx512skx-i64x4 target would allow users of double precision arithmetic (and other 64 bit types) to have significant performance gains at minimal frequency penalty on skylake. It would also allow users of float (and other 32 bit types) to have a larger pool of registers to access with zero frequency penalty.

With the addition of these new targets, it may also be useful to expand the existing documentation on the targets to address a few subtle bits:

(1) Why are there separate avx2-i32x4 and avx2-i64x4 targets?
(2) Why are there not separate targets for avx512skx-i32x4 and avx512skx-i64x4?
(3) When given various options for "base type" and "program count" on a target architecture, how does one select a target?

**Comment by mindbeast** [comment]
Pulled from discussion [here](https://github.com/ispc/ispc/discussions/2170).

**Comment by mindbeast** [comment]
Exploring this a little further, I was somewhat surprised to see that with the combination of a few flags to ispc, I can actually get half of what I already want.

-O3 --target=avx2-x4 --cpu=skx

The above seems to allow register usage of the xmm16 - xmm31, but without some of the avx512 specific instructions.

**Comment by dbabokin** [comment]
Correct, through avx512 specific instruction would include some forms for gather/scatters, which is quite unfortunate.

**Comment by dbabokin** [comment]
`avx512skx-i32x4` is mostly ready, all functional tests are passing. Please give it a try:
Windows build: https://ci.appveyor.com/project/dbabokin/ispc/builds/40966197/job/wb9na74jc8km6xse/artifacts
Linux build: https://ci.appveyor.com/project/dbabokin/ispc/builds/40966197/job/ay16064ago2yd734/artifacts
